### PR TITLE
feat(grouping): Show class basename for Java

### PR DESCRIPTION
**Pull request by untitaker**
Java method names on their own are not useful enough to describe what
the issue is about.

Fix INGEST-254

### CODE REVIEW

File: src/sentry/eventtypes/base.py [hunk 1]
```diff
@@ -18,6 +18,14 @@ def format_title_from_tree_label(tree_label):
         else:
             part = x.get("function") or x.get("package") or x.get("type")
 
+            classbase = x.get("classbase")
```
Comment (jjbayer) [inline src/sentry/eventtypes/base.py:21]: nit: Now that this loop body is getting longer, should we rename `x`?

File: src/sentry/grouping/strategies/newstyle.py [hunk 2]
```diff
@@ -364,7 +369,16 @@ def frame(frame, event, context, **meta):
         rv.update(contributes=False, hint="ignored due to recursion")
 
     if rv.tree_label:
-        rv.tree_label = {"datapath": frame.datapath, **rv.tree_label}
+        tree_label = {}
```
Comment (jjbayer) [inline src/sentry/grouping/strategies/newstyle.py:372]: Why is this change necessary?

File: tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/mobile@2021_02_12/node_exception_weird.pysnap [hunk 3]
```diff
@@ -135,7 +135,7 @@ app-depth-4:
 --------------------------------------------------------------------------
 app-depth-5:
   hash: "d50c59998b650da06b01b9f2b8ecc2f7"
-  tree_label: "eventFromException | captureException | invokeClient | captureException | <anonymous>"
+  tree_label: "eventFromException | captureException | invokeClient | captureException"
```
Comment (jjbayer) [inline tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/mobile@2021_02_12/node_exception_weird.pysnap:138]: I guess it's good that this noise is removed, but what change caused its disappearance?

File: src/sentry/grouping/strategies/newstyle.py (hunk 2 above)
Comment (untitaker) [reply src/sentry/grouping/strategies/newstyle.py:372]: grouping components by default take the first tree label from their child component and propagate it. In the case of a frame this meant that only one tree-label from any of the frame's child components was propagated outside, so there can be only either `function` or `package` inside. (`_calculate_tree_label` in grouping/component.py)

This change here basically overrides the default tree label of `rv.tree_label = rv.values[0].tree_label` to instead merge all values's dictionaries into one.

This also causes the weirdness in the snapshots you pointed out because we used to use the function component's tree label even if the component was non-contributing (for some reason I haven't figured out, but that was a bug for sure)

(Note: Ideally at some point I'd like to refactor this entire mess such that tree labels are not propagated the way they are, I think I'll do that in a followup PR

File: src/sentry/eventtypes/base.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/eventtypes/base.py:21]: I refactored the entire thing to be closer to how JS is structured.

File: src/sentry/grouping/strategies/newstyle.py (hunk 2 above)
Comment (jjbayer) [reply src/sentry/grouping/strategies/newstyle.py:372]: Makes sense.

**Review by jjbayer — APPROVED** [review]
